server/server.js

```diff
@@ -22,8 +22,122 @@ app.use(express.static(reactStaticDir));
 app.use(express.static(uploadsStaticDir));
 app.use(express.json());
 
-app.get('/api/hello', (req, res) => {
-  res.json({ message: 'Hello World!' });
+app.get('/api/restaurants', async (req, res) => {
+  try {
+    const sql = `
+    select *
+      from "restaurants"
+    `;
+    const result = await db.query(sql);
+    res.json(result.rows);
+  } catch (err) {
+    console.error(err);
+    res.status(500).json({error: 'an unexpected error occured'});
+  }
+});
+
+app.get('/api/restaurants/:id', async (req, res) => {
+  try {
+    const id = Number(req.params.id);
+    if (!Number.isInteger(id) || id < 1) {
+      res.status(400).json({error: 'id must be a positive integer'});
+      return;
+    }
+    const sql = `
+    select * from restaurants
+      where "id" = $1
+    `;
+    const params = [id];
+    const result = await db.query(sql, params);
+    if(!result.rows[0]) {
+      res.status(400).json({error: 'this id does not exist'});
+      return;
+    }
+    res.json(result.rows[0]);
+  } catch(err) {
+    console.error(err);
+    res.status(500).json({error: 'an unexpected error has occured'});
+  }
+});
+
+app.post('/api/restaurants', async (req, res) => {
+  try {
+    const { name, location, priceRange } = req.body;
+    if (!name || !location || !priceRange) {
+      res.status(400).json({error: 'name, location, and priceRange are required'});
+      return;
+    }
+    if (typeof name !== 'string' || typeof location !== 'string' || typeof priceRange !== 'number') {
+      res.status(400).json({error: 'invalid input on either name, location, or priceRange'});
+      return;
+    }
+    const sql = `
+    insert into "restaurants" ("name", "location", "priceRange")
+      values ($1, $2, $3)
+      returning *
+    `;
+    const params = [name, location, priceRange];
+    const result = await db.query(sql, params);
+    res.json(result.rows[0]);
+  } catch (err) {
+    console.error(err);
+    res.status(500).json({error: 'an unexpected error has occured'});
+  }
+});
+
+app.patch('/api/restaurants/:id', async (req, res) => {
+  try {
+    const {name, location, priceRange} = req.body;
+    if (typeof name !== 'string' || typeof location !== 'string' || typeof priceRange !== 'number') {
+      res.status(400).json({ error: 'name, location, or priceRange have invalid inputs' });
+      return;
+    }
+    const id = Number(req.params.id);
+    if(!Number.isInteger(id) || id < 1) {
+      res.status(400).json({error: 'id must be a positive integer'});
+      return;
+    }
+    const sql = `
+    update "restaurants"
```

Comment:
This is updating the entire restaurants record, so you probably want to use `app.put` instead of `app.patch`. That would be the more common way to design the API.